lib/core/test/unit/Cardano/Pool/DB/Arbitrary.hs

```diff
@@ -371,3 +378,16 @@ instance Arbitrary StakePoolsFixture where
         appendPair pools pairs slot = do
             pool <- elements pools
             return $ (pool,slot):pairs
+
+instance Arbitrary URI where
+    arbitrary = elements
+        [fromJust (parseURI "https://my.little.friend")
+        ,fromJust (parseURI "http://its-friday.com:8000")]
+
+instance Arbitrary PoolMetadataSource where
+    arbitrary = oneof [ pure None
```

Comment:
No problem using `genericArbitrary` / `genericShrink` here.

Reply:
I'm not sure if that will output valid URIs. And it requires a lot of orphan instances.